### PR TITLE
Fix #11133 - Removing buf_hex to

### LINKED ISSUE #11133: Kill buf_hex from RAsmCode and RAsmOp
Kill buf_hex from RAsmCode and RAsmOp

### PR DESCRIPTION
**Pull request by wargio**
**Wait all the tests to be green before mergin**

### CODE REVIEW

**Comment by wargio** [comment]
https://github.com/radare/radare2/issues/11133

**Comment by wargio** [comment]
Everything is broken!! YEAH

File: libr/main/rasm2.c [hunk 1]
```diff
@@ -448,7 +450,9 @@ static int rasm_asm(RAsmState *as, const char *buf, ut64 offset, ut64 len, int b
 					}
 					printf ("\n");
 				} else {
-					print_buf (as, acode->buf_hex);
+					char* str = r_asm_code_get_hex (acode);
+					print_buf (as, str);
```
Comment (radare) [inline libr/main/rasm2.c]: check for null

File: libr/main/rasm2.c [hunk 2]
```diff
@@ -391,6 +390,9 @@ static int rasm_disasm(RAsmState *as, char *buf, ut64 offset, int len, int bits,
 
 static void print_buf(RAsmState *as, char *str) {
 	int i;
+	if (!str) {
```
Comment (radare) [inline libr/main/rasm2.c]: r_return_if_fail

File: libr/core/cmd_write.c [hunk 3]
```diff
@@ -1501,11 +1505,12 @@ static int cmd_write(void *data, const char *input) {
 				r_asm_set_pc (core->assembler, core->offset);
 				RAsmCode *acode = r_asm_assemble_file (core->assembler, file);
 				if (acode) {
-					if (input[2] == '*') {
-						cmd_write_hexpair (core, acode->buf_hex);
+					char* hex = r_asm_code_get_hex (acode);
+					if (input[2] == '*' && hex) {
```
Comment (radare) [inline libr/core/cmd_write.c]: the hex check have nothing to do with '*'

File: libr/core/cmd_write.c [hunk 4]
```diff
@@ -1479,9 +1481,11 @@ static int cmd_write(void *data, const char *input) {
 						}
 						if (*b) {
 							RAsmCode *ac = r_asm_massemble (core->assembler, b);
-							if (ac && *ac->buf_hex) {
-								r_cons_printf ("wx %s @ 0x%08"PFMT64x"\n",  ac->buf_hex, addr);
+							char* hex = r_asm_code_get_hex (ac);
+							if (hex) {
+								r_cons_printf ("wx %s @ 0x%08"PFMT64x"\n",  hex, addr);
 							}
```
Comment (radare) [inline libr/core/cmd_write.c]: move free(hex) inside the if(hex)

File: libr/asm/op.c [hunk 5]
```diff
@@ -22,13 +22,14 @@ R_API void r_asm_op_init(RAsmOp *op) {
...
 	r_strbuf_fini (&op->buf_asm);
-	r_strbuf_fini (&op->buf_hex);
 	r_buf_fini (op->buf_inc);
 }
 
 // accessors
 R_API char *r_asm_op_get_hex(RAsmOp *op) {
-	return r_strbuf_get (&op->buf_hex);
+	char* str = calloc (op->size, 2);
+	r_hex_bin2str (r_strbuf_get (&op->buf), op->size, str);
```
Comment (radare) [inline libr/asm/op.c]: check for null

File: libr/asm/code.c [hunk 6]
```diff
@@ -64,3 +63,14 @@ R_API char *r_asm_code_equ_replace (RAsmCode *code, char *str) {
 	}
 	return str;
 }
+
+R_API char* r_asm_code_get_hex(RAsmCode *acode) {
+	if (!acode) {
```
Comment (radare) [inline libr/asm/code.c]: r_return_val_if_fail

File: libr/main/rasm2.c (hunk 1 above)
Comment (wargio) [reply libr/main/rasm2.c]: no need. checked inside `print_buf`

Comment (radare) [reply libr/main/rasm2.c]: static functions should have no null pointer checks for the arguments coming from the caller. tahts the rule we decided to use

**Comment by radare** [comment]
.

**Comment by radare** [comment]
Screenshot 2019-03-22 at 17 21 43

**Comment by wargio** [comment]
When green, then can be merged :)

**Comment by radare** [comment]
Thanks!

**Pull request merged**